random1on1/api/channels.py

```diff
@@ -1,36 +1,132 @@
+import json
 from abc import ABC
 from abc import abstractmethod
+from datetime import datetime
+from functools import reduce
 
-from discord.channel import GroupChannel
-from discord.channel import TextChannel
+from discord import CategoryChannel
+from discord import GroupChannel
+from discord import Role
+from discord import TextChannel
+from networkx import connected_components
+from networkx import union
+from preconditions import preconditions
 
-from random1on1.api.participant import Participant
+from .pairings import Pairings
 
 
-class ReadWriteChannel(ABC, TextChannel):
+class Random1on1Channel(ABC):
 
+    @preconditions(lambda name: len(name) > 0)
     @abstractmethod
-    def __init__(self, channel_name: str):
-        # TODO: Add precondition checks here
-        self.channel_name = channel_name
+    def __init__(self, name: str, category: CategoryChannel):
+        self.name = name
+        channels = [c for c in category.text_channels if c.name == self.name]
+        if len(channels) == 0:
+            channel = category.create_text_channel(name=self.name)
+        elif len(channels) == 1:
+            channel = channels[0]
+        else:
+            raise RuntimeError(
+                f"Found multiple channels in category {category.name} with reserved channel name {self.name}"
+            )
```

Comment:
```
def check_channel(channel_name, category):
    """Checks if channel exists in the categorie provided and creates it otherwise."""

    channel_count = list(category.text_channels).count(channel_name)
    if channel_count == 0:
        channel = category.create_text_channel(name=self.name)
    if channel_count >= 1:
        raise RuntimeError(
                f"Found multiple channels in category {category.name} with reserved channel name {self.name}"
            )
```

Reply:
The code you put here would alway give you a count of zero because 
1. `category.text_channels` is a `list[TextChannel]`, so you're running count on `list[list[TextChannel]]`
2. Even if you did `category.text_channels.count(channel_name)` you would always get `0` because `channel_name` is of type `str` not `TextChannel`

You cannot convert `channel_name` into a `TextChannel` and then count instances because every channel is assigned a unique value for `id` and you would just get `0` because none of the channels have the same identifier. You _must_ do a filter with comparison just by channel name unfortunately. 

Also worth noting, that we want to allow for the case where `channel_count == 1`. This channel check needs to be done every time the discord bot connects. On the first connection it might have to create the channel, but after the first connection the channel should already exist, so we should just attach to that existing channel for continuity's sake.